pypeit/edgetrace.py

```diff
@@ -250,10 +251,9 @@ class EdgeTraceSet(DataContainer):
 
     Args:
         traceimg (:class:`pypeit.images.buildcalibration.TraceImage`):
```

Comment:
This class needs to be able to run on a normal image and not just an pypeit.images.buildcalibration.TraceImage object. I think this is why bpm has been removed now as an input, because you are now getting the bpm out of the image object BPM. That is fine, but please reinstate the original functionality. 

This is a core operation, i.e. finding slit edges. It should run on an image, not a complex object. If you need other
complex things make them keyword arguments set to appropriate defaults, or made decisions based on what is passed in, but an image always need to a possible input.

Reply:
I disagree.  If you look back at `pypeit/calibrations.py` you'll see that this usage is consistent with our other main modules.  E.g., `EdgeTraceSet`, `BuildWaveCalib`, `FlatField`, etc, all operate on `PypeItImage` instances, not simple arrays.  If people want to trace a set files, they can use (or follow) the `pypeit_trace_edges` script.

Reply:
I see that you are correct about this being consistent with BuildWaveCalib and FlatField. I somehow missed those modifications go by. I'm going to drop this point in the context of this PR, since it touches on a bigger issue which now affects nearly all of our classes. But in my opinion, the approach that we are adopting of writing code that is inseparable from complex PypeIt objects is a very bad one. I tolerated passing in spectrograph and par because they are easy to instantiate in one line of code. That is far from true for a pypeitimage object. The only thing that making the object an argument saves you over making the image the argument is a few additional arguments in the argument list, which is hardly a good tradeoff in my opinion. I can think of many cases where I can read in an image and create a spectrograph class, but where I cannot create a pypeitimage object or run the pypeit_trace_edges script.